content/OTP/OTPs_Explained.adoc

```diff
@@ -23,3 +23,69 @@ Copying an OTP will not allow another user to spoof a YubiKey -- the counter val
 will allow the validation server to know which OTPs have already been used.
 
 image:otp_details.png[]
+
+== The Yubico OTP generation algorithm
+The YubiKey OTP generation is made up of the following fields, encrypted with a unique AES-128 bit key. The result is the 32 character modhex string included after the 12 character public ID.
+
+|===
+|Mnemonic |Byte offset |Size |Description
+|uid 
+|0 
+|6 
+|Private (secret) id
+
+|useCtr 
+|6 
+|2 
+|Usage counter
+
+|tstp 
+|8 
+|3 
+|Timestamp
+
+|sessionCtr 
+|11 
+|1 
+|Session usage counter
+
+|rnd 
+|12 
+|2 
+|Random number
+
+|crc 14 2 CRC16 checksum
+|===
+
+
+=== Private ID
+The private id field comprises 6 bytes copied from the private id field configuration value. This field can be used to store a private identity if shared encryption keys are used.
```

Comment:
What does "if shared encryption keys are used" refer to in this context? Shared between which parties?

Reply:
Bump

Reply:
@emlun why bump? he did update the docs, see the comment below by Yubi-David

Reply:
Sorry, I meant to bump @dain :slightly_smiling_face: